Revision decreasing after panic during compaction

### Bug report criteria

- [x] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [x] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [x] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [x] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

Failure in https://github.com/etcd-io/etcd/actions/runs/8659974818

![image](https://github.com/etcd-io/etcd/assets/5873433/8508b160-c2f7-49f1-85b8-9930a1132826)

### What did you expect to happen?

Revision doesn't decrease

### How can we reproduce it (as minimally and precisely as possible)?

Follow https://github.com/etcd-io/etcd/tree/main/tests/robustness#re-evaluate-existing-report to validate report from https://github.com/etcd-io/etcd/actions/runs/8659974818

Run `TestRobustnessExploratory/Kubernetes/LowTraffic/ClusterOfSize1` with failpoint `compactBeforeSetFinishedCompact=panic()`

### Anything else we need to know?

TODO:
* [x] Try to reproduce the issue
* [x] Try to reproduce the issue on amd64

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# paste output here

$ etcdctl version
# paste output here
```

</details>


### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>


### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>


### Relevant log output

_No response_

FYI, Reproduce it with https://github.com/etcd-io/etcd/pull/17782 after 17160.566s (8vcore, 16cpus, arm64, 5.15.0-1060-azure). 

```go
    logger.go:146: 2024-04-14T15:40:19.770Z     INFO    started server. {"name": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0", "pid": 224554}
    logger.go:146: 2024-04-14T15:40:19.770Z     INFO    Verifying cluster health after failpoint        {"failpoint": "compactBeforeSetFinishedCompact=panic()"}
    logger.go:146: 2024-04-14T15:40:20.771Z     INFO    Finished injecting failures
    logger.go:146: 2024-04-14T15:40:21.852Z     INFO    Recorded operations     {"operations": 937, "successRate": 0.7257203842049093}
    logger.go:146: 2024-04-14T15:40:21.852Z     INFO    Traffic from successful requests        {"qps": 266.3974175431209, "operations": 680, "period": "2.552577297s"}
    logger.go:146: 2024-04-14T15:40:21.852Z     INFO    Finished simulating traffic     {"max-revision": 415}
    logger.go:146: 2024-04-14T15:40:21.853Z     INFO    killing server...       {"name": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0"}
    logger.go:146: 2024-04-14T15:40:21.853Z     INFO    stopping server...      {"name": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0"}
    logger.go:146: 2024-04-14T15:40:21.856Z     INFO    stopped server. {"name": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0"}
    logger.go:146: 2024-04-14T15:40:21.858Z     INFO    Validating linearizable operations      {"timeout": "5m0s"}
    logger.go:146: 2024-04-14T15:40:25.004Z     ERROR   Linearization failed    {"duration": "3.145713744s"}
    validate.go:36: Failed linearization, skipping further validation
    logger.go:146: 2024-04-14T15:40:25.004Z     INFO    Saving robustness test report   {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1"}
    logger.go:146: 2024-04-14T15:40:25.004Z     INFO    Saving member data dir  {"member": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0", "path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/server-TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0"}
    logger.go:146: 2024-04-14T15:40:25.004Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-1/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.004Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-2/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.005Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-3/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.005Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-3/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.006Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-4/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.006Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-4/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.007Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-5/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.007Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-5/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.008Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-6/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.009Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-6/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.009Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-7/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.010Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-7/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.010Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-8/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.011Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-8/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.011Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-9/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.012Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-9/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.012Z     INFO    Saving watch operations {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-10/watch.json"}
    logger.go:146: 2024-04-14T15:40:25.013Z     INFO    Saving operation history        {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/client-10/operations.json"}
    logger.go:146: 2024-04-14T15:40:25.013Z     INFO    Saving visualization    {"path": "/tmp/results/TestRobustnessExploratory_Kubernetes_LowTraffic_ClusterOfSize1/history.html"}
    logger.go:146: 2024-04-14T15:40:25.037Z     INFO    closing test cluster...
    logger.go:146: 2024-04-14T15:40:25.037Z     INFO    closing server...       {"name": "TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize1-test-0"}
    logger.go:146: 2024-04-14T15:40:25.037Z     INFO    removing directory      {"data-dir": "/tmp/TestRobustnessExploratoryKubernetesLowTrafficClusterOfSize13457912620/001"}
    logger.go:146: 2024-04-14T15:40:25.037Z     INFO    closed test cluster.
--- FAIL: TestRobustnessExploratory (7.4FAIL: (code:1):
  % (cd tests && 'env' 'ETCD_VERIFY=all' 'go' 'test' 'go.etcd.io/etcd/tests/v3/robustness' '-timeout=30m' '-v' '--count' '10000' '--timeout' '2000m' '--failfast' '--run' 'TestRobustnessExploratory/Kubernetes/LowTraffic/ClusterOfSize1')
9s)
    --- FAIL: TestRobustnessExploratory/Kubernetes/LowTraffic/ClusterOfSize1 (7.48s)
FAIL
FAIL    go.etcd.io/etcd/tests/v3/robustness     17160.566s
FAIL

```
Still checking it.

--- a/tests/robustness/failpoint/gofail.go
+++ b/tests/robustness/failpoint/gofail.go
@@ -31,36 +31,37 @@ import (
 )
 
 var (
-	DefragBeforeCopyPanic                    Failpoint = goPanicFailpoint{"defragBeforeCopy", triggerDefrag{}, AnyMember}
-	DefragBeforeRenamePanic                  Failpoint = goPanicFailpoint{"defragBeforeRename", triggerDefrag{}, AnyMember}
-	BeforeCommitPanic                        Failpoint = goPanicFailpoint{"beforeCommit", nil, AnyMember}
-	AfterCommitPanic                         Failpoint = goPanicFailpoint{"afterCommit", nil, AnyMember}
-	RaftBeforeSavePanic                      Failpoint = goPanicFailpoint{"raftBeforeSave", nil, AnyMember}
-	RaftAfterSavePanic                       Failpoint = goPanicFailpoint{"raftAfterSave", nil, AnyMember}
-	BackendBeforePreCommitHookPanic          Failpoint = goPanicFailpoint{"commitBeforePreCommitHook", nil, AnyMember}
-	BackendAfterPreCommitHookPanic           Failpoint = goPanicFailpoint{"commitAfterPreCommitHook", nil, AnyMember}
-	BackendBeforeStartDBTxnPanic             Failpoint = goPanicFailpoint{"beforeStartDBTxn", nil, AnyMember}
-	BackendAfterStartDBTxnPanic              Failpoint = goPanicFailpoint{"afterStartDBTxn", nil, AnyMember}
-	BackendBeforeWritebackBufPanic           Failpoint = goPanicFailpoint{"beforeWritebackBuf", nil, AnyMember}
-	BackendAfterWritebackBufPanic            Failpoint = goPanicFailpoint{"afterWritebackBuf", nil, AnyMember}
-	CompactBeforeCommitScheduledCompactPanic Failpoint = goPanicFailpoint{"compactBeforeCommitScheduledCompact", triggerCompact{}, AnyMember}
-	CompactAfterCommitScheduledCompactPanic  Failpoint = goPanicFailpoint{"compactAfterCommitScheduledCompact", triggerCompact{}, AnyMember}
-	CompactBeforeSetFinishedCompactPanic     Failpoint = goPanicFailpoint{"compactBeforeSetFinishedCompact", triggerCompact{}, AnyMember}
-	CompactAfterSetFinishedCompactPanic      Failpoint = goPanicFailpoint{"compactAfterSetFinishedCompact", triggerCompact{}, AnyMember}
-	CompactBeforeCommitBatchPanic            Failpoint = goPanicFailpoint{"compactBeforeCommitBatch", triggerCompact{multiBatchCompaction: true}, AnyMember}
-	CompactAfterCommitBatchPanic             Failpoint = goPanicFailpoint{"compactAfterCommitBatch", triggerCompact{multiBatchCompaction: true}, AnyMember}
-	RaftBeforeLeaderSendPanic                Failpoint = goPanicFailpoint{"raftBeforeLeaderSend", nil, Leader}
-	RaftBeforeFollowerSendPanic              Failpoint = goPanicFailpoint{"raftBeforeFollowerSend", nil, Follower}
-	RaftBeforeApplySnapPanic                 Failpoint = goPanicFailpoint{"raftBeforeApplySnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	RaftAfterApplySnapPanic                  Failpoint = goPanicFailpoint{"raftAfterApplySnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	RaftAfterWALReleasePanic                 Failpoint = goPanicFailpoint{"raftAfterWALRelease", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	RaftBeforeSaveSnapPanic                  Failpoint = goPanicFailpoint{"raftBeforeSaveSnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	RaftAfterSaveSnapPanic                   Failpoint = goPanicFailpoint{"raftAfterSaveSnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	ApplyBeforeOpenSnapshot                  Failpoint = goPanicFailpoint{"applyBeforeOpenSnapshot", triggerBlackhole{waitTillSnapshot: true}, Follower}
-	BeforeApplyOneConfChangeSleep            Failpoint = killAndGofailSleep{"beforeApplyOneConfChange", time.Second}
-	RaftBeforeSaveSleep                      Failpoint = gofailSleepAndDeactivate{"raftBeforeSave", time.Second}
-	RaftAfterSaveSleep                       Failpoint = gofailSleepAndDeactivate{"raftAfterSave", time.Second}
-	SleepBeforeSendWatchResponse             Failpoint = gofailSleepAndDeactivate{"beforeSendWatchResponse", time.Second}
+	DefragBeforeCopyPanic                     Failpoint = goPanicFailpoint{"defragBeforeCopy", triggerDefrag{}, AnyMember}
+	DefragBeforeRenamePanic                   Failpoint = goPanicFailpoint{"defragBeforeRename", triggerDefrag{}, AnyMember}
+	BeforeCommitPanic                         Failpoint = goPanicFailpoint{"beforeCommit", nil, AnyMember}
+	AfterCommitPanic                          Failpoint = goPanicFailpoint{"afterCommit", nil, AnyMember}
+	RaftBeforeSavePanic                       Failpoint = goPanicFailpoint{"raftBeforeSave", nil, AnyMember}
+	RaftAfterSavePanic                        Failpoint = goPanicFailpoint{"raftAfterSave", nil, AnyMember}
+	BackendBeforePreCommitHookPanic           Failpoint = goPanicFailpoint{"commitBeforePreCommitHook", nil, AnyMember}
+	BackendAfterPreCommitHookPanic            Failpoint = goPanicFailpoint{"commitAfterPreCommitHook", nil, AnyMember}
+	BackendBeforeStartDBTxnPanic              Failpoint = goPanicFailpoint{"beforeStartDBTxn", nil, AnyMember}
+	BackendAfterStartDBTxnPanic               Failpoint = goPanicFailpoint{"afterStartDBTxn", nil, AnyMember}
+	BackendBeforeWritebackBufPanic            Failpoint = goPanicFailpoint{"beforeWritebackBuf", nil, AnyMember}
+	BackendAfterWritebackBufPanic             Failpoint = goPanicFailpoint{"afterWritebackBuf", nil, AnyMember}
+	CompactBeforeCommitScheduledCompactPanic  Failpoint = goPanicFailpoint{"compactBeforeCommitScheduledCompact", triggerCompact{}, AnyMember}
+	CompactAfterCommitScheduledCompactPanic   Failpoint = goPanicFailpoint{"compactAfterCommitScheduledCompact", triggerCompact{}, AnyMember}
+	CompactBeforeSetFinishedCompactPanic      Failpoint = goPanicFailpoint{"compactBeforeSetFinishedCompact", triggerCompact{}, AnyMember}
+	BatchCompactBeforeSetFinishedCompactPanic Failpoint = goPanicFailpoint{"compactBeforeSetFinishedCompact", triggerCompact{multiBatchCompaction: true}, AnyMember}
+	CompactAfterSetFinishedCompactPanic       Failpoint = goPanicFailpoint{"compactAfterSetFinishedCompact", triggerCompact{}, AnyMember}
+	CompactBeforeCommitBatchPanic             Failpoint = goPanicFailpoint{"compactBeforeCommitBatch", triggerCompact{multiBatchCompaction: true}, AnyMember}
+	CompactAfterCommitBatchPanic              Failpoint = goPanicFailpoint{"compactAfterCommitBatch", triggerCompact{multiBatchCompaction: true}, AnyMember}
+	RaftBeforeLeaderSendPanic                 Failpoint = goPanicFailpoint{"raftBeforeLeaderSend", nil, Leader}
+	RaftBeforeFollowerSendPanic               Failpoint = goPanicFailpoint{"raftBeforeFollowerSend", nil, Follower}
+	RaftBeforeApplySnapPanic                  Failpoint = goPanicFailpoint{"raftBeforeApplySnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	RaftAfterApplySnapPanic                   Failpoint = goPanicFailpoint{"raftAfterApplySnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	RaftAfterWALReleasePanic                  Failpoint = goPanicFailpoint{"raftAfterWALRelease", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	RaftBeforeSaveSnapPanic                   Failpoint = goPanicFailpoint{"raftBeforeSaveSnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	RaftAfterSaveSnapPanic                    Failpoint = goPanicFailpoint{"raftAfterSaveSnap", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	ApplyBeforeOpenSnapshot                   Failpoint = goPanicFailpoint{"applyBeforeOpenSnapshot", triggerBlackhole{waitTillSnapshot: true}, Follower}
+	BeforeApplyOneConfChangeSleep             Failpoint = killAndGofailSleep{"beforeApplyOneConfChange", time.Second}
+	RaftBeforeSaveSleep                       Failpoint = gofailSleepAndDeactivate{"raftBeforeSave", time.Second}
+	RaftAfterSaveSleep                        Failpoint = gofailSleepAndDeactivate{"raftAfterSave", time.Second}
+	SleepBeforeSendWatchResponse              Failpoint = gofailSleepAndDeactivate{"beforeSendWatchResponse", time.Second}
 )
 
 type goPanicFailpoint struct {

--- a/tests/robustness/makefile.mk
+++ b/tests/robustness/makefile.mk
@@ -44,6 +44,11 @@ test-robustness-issue17529: /tmp/etcd-v3.5.12-beforeSendWatchResponse/bin
 	GO_TEST_FLAGS='-v --run=TestRobustnessRegression/Issue17529 --count 100 --failfast --bin-dir=/tmp/etcd-v3.5.12-beforeSendWatchResponse/bin' $(MAKE) test-robustness && \
 	 echo "Failed to reproduce" || echo "Successful reproduction"
 
+.PHONY: test-robustness-issue17780
+test-robustness-issue17780: /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/bin
+	GO_TEST_FLAGS='-v --run=TestRobustnessRegression/Issue17780 --count 200 --failfast --bin-dir=/tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/bin' make test-robustness && \
+	  echo "Failed to reproduce" || echo "Successful reproduction"
+
 # Failpoints
 
 GOPATH = $(shell go env GOPATH)
@@ -112,6 +117,20 @@ $(GOPATH)/bin/gofail: tools/mod/go.mod tools/mod/go.sum
 	cd /tmp/etcd-v3.5.12-beforeSendWatchResponse/; \
 	  patch -l server/etcdserver/api/v3rpc/watch.go ./beforeSendWatchResponse/watch.patch; \
 	  patch -l build.sh ./beforeSendWatchResponse/build.patch; \
+	  go get go.etcd.io/gofail@${GOFAIL_VERSION}; \
+	  (cd server; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \
+	  (cd etcdctl; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \
+	  (cd etcdutl; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \
+	  (cd tools/mod; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \
+	  FAILPOINTS=true ./build;
+
+/tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/bin: $(GOPATH)/bin/gofail
+	rm -rf /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/
+	mkdir -p /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/
+	git clone --depth 1 --branch v3.5.13 https://github.com/etcd-io/etcd.git /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/
+	cp -r ./tests/robustness/patches/compactBeforeSetFinishedCompact /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/
+	cd /tmp/etcd-v3.5.13-compactBeforeSetFinishedCompact/; \
+	  patch -l server/mvcc/kvstore_compaction.go ./compactBeforeSetFinishedCompact/kvstore_compaction.patch; \
 	  go get go.etcd.io/gofail@${GOFAIL_VERSION}; \
 	  (cd server; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \
 	  (cd etcdctl; go get go.etcd.io/gofail@${GOFAIL_VERSION}); \

--- a/tests/robustness/patches/compactBeforeSetFinishedCompact/kvstore_compaction.patch
+++ b/tests/robustness/patches/compactBeforeSetFinishedCompact/kvstore_compaction.patch
@@ -1,0 +1,26 @@
+From 6b034466aa0ac2b46fe01fb5bd2233946f46d453 Mon Sep 17 00:00:00 2001
+From: Wei Fu <fuweid89@gmail.com>
+Date: Wed, 24 Apr 2024 12:14:27 +0800
+Subject: [PATCH] server/mvcc: introduce compactBeforeSetFinishedCompact
+ failpoint
+
+Signed-off-by: Wei Fu <fuweid89@gmail.com>
+---
+ server/mvcc/kvstore_compaction.go | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/server/mvcc/kvstore_compaction.go b/server/mvcc/kvstore_compaction.go
+index c7d343d5c..89defbd9e 100644
+--- a/server/mvcc/kvstore_compaction.go
++++ b/server/mvcc/kvstore_compaction.go
+@@ -59,6 +59,7 @@ func (s *store) scheduleCompaction(compactMainRev, prevCompactRev int64) (KeyVal
+ 		}
+ 
+ 		if len(keys) < s.cfg.CompactionBatchLimit {
++			// gofail: var compactBeforeSetFinishedCompact struct{}
+ 			rbytes := make([]byte, 8+1+8)
+ 			revToBytes(revision{main: compactMainRev}, rbytes)
+ 			tx.UnsafePut(buckets.Meta, finishedCompactKeyName, rbytes)
+-- 
+2.34.1
+

--- a/tests/robustness/scenarios/scenarios.go
+++ b/tests/robustness/scenarios/scenarios.go
@@ -210,6 +210,19 @@ func Regression(t *testing.T) []TestScenario {
 			options.WithSnapshotCount(100),
 		),
 	})
+
+	scenarios = append(scenarios, TestScenario{
+		Name:      "Issue17780",
+		Profile:   traffic.LowTraffic.WithoutCompaction(),
+		Failpoint: failpoint.BatchCompactBeforeSetFinishedCompactPanic,
+		Traffic:   traffic.Kubernetes,
+		Cluster: *e2e.NewConfig(
+			e2e.WithClusterSize(1),
+			e2e.WithCompactionBatchLimit(300),
+			e2e.WithSnapshotCount(1000),
+			e2e.WithGoFailEnabled(true),
+		),
+	})
 	if v.Compare(version.V3_5) >= 0 {
 		opts := []e2e.EPClusterOption{
 			e2e.WithSnapshotCount(100),

